runtime/vm/UpcallVMHelpers.cpp

```diff
@@ -34,6 +34,19 @@ extern "C" {
 
 #if JAVA_SPEC_VERSION >= 16
 
+extern void c_cInterpreter(J9VMThread *currentThread);
+extern bool buildCallInStackFrameHelper(J9VMThread *currentThread, J9VMEntryLocalStorage *newELS, bool returnsObject);
+extern void restoreCallInFrameHelper(J9VMThread *currentThread);
+
+static UDATA JNICALL native2InterpJavaUpcallImpl(J9UpcallMetaData *data, void *argsListPointer);
```

Comment:
The return type should probably be `U_64` to be clear that values are up to 64 bits wide.

Reply:
We might need to support zOS 32bit in the future, in which case it should remain as `UDATA` at this point.

Reply:
Even if, or when, we add support for a 32-bit platform, this should be `U_64` to support `double` and `long` return values.

Reply:
PR is created at https://github.com/eclipse-openj9/openj9/pull/15900 for the fixes.